st2api/st2api/controllers/exp/inquiries.py

```diff
@@ -0,0 +1,273 @@
+    # for the DB layer
+    model = InquiryAPI
+    access = ActionExecution
+
+    def get_all(self, requester_user=None, limit=None, **raw_filters):
+        """Retrieve multiple Inquiries
+
+            Handles requests:
+                GET /inquiries/
+        """
+
+        raw_inquiries = super(InquiriesController, self)._get_all(
+            limit=limit,
+            raw_filters={
+                'status': action_constants.LIVEACTION_STATUS_PENDING,
+                'runner': INQUIRY_RUNNER
+            }
+        )
+
+        # Transform to InquiryResponseAPI model
+        inquiries = [InquiryResponseAPI.from_inquiry_api(InquiryAPI.from_dict(raw_inquiry))
+                     for raw_inquiry in json.loads(raw_inquiries.body)]
+
+        # Repackage into Response with correct headers
+        resp = Response(json=inquiries)
+        resp.headers['X-Total-Count'] = raw_inquiries.headers['X-Total-Count']
+        if limit:
+            resp.headers['X-Limit'] = str(limit)
+        return resp
+
+    def get_one(self, inquiry_id, requester_user=None):
+        """Retrieve a single Inquiry
+
+            Handles requests:
+                GET /inquiries/<inquiry id>
+        """
+
+        # Retrieve the desired inquiry
+        #
+        # (Passing permission_type here leverages _get_one_by_id's built-in
+        # RBAC assertions)
+        inquiry = self._get_one_by_id(
+            id=inquiry_id,
+            requester_user=requester_user,
+            permission_type=PermissionType.INQUIRY_VIEW
+        )
+
+        if not self._inquiry_sanity_check(inquiry):
+            return
+
+        return InquiryResponseAPI.from_inquiry_api(inquiry)
+
+    def put(self, inquiry_id, response_data, requester_user):
+        """Provide response data to an Inquiry
+
+            In general, provided the response data validates against the provided
+            schema, and the user has the appropriate permissions to respond,
+            this will set the Inquiry execution to a successful status, and resume
+            the parent workflow.
+
+            Handles requests:
+                PUT /inquiries/<inquiry id>
+        """
+
+        LOG.debug("Inquiry %s received response payload: %s" % (inquiry_id, response_data.response))
+
+        # Retrieve details of the inquiry via ID (i.e. params like schema)
+        inquiry = self._get_one_by_id(
+            id=inquiry_id,
+            requester_user=requester_user,
+            permission_type=PermissionType.INQUIRY_RESPOND
+        )
+
+        if not self._inquiry_sanity_check(inquiry):
+            return
+
+        if not requester_user:
+            requester_user = UserDB(cfg.CONF.system_user.user)
+
+        # Determine permission of this user to respond to this Inquiry
+        if not self._can_respond(inquiry, requester_user):
+            abort(
+                http_client.FORBIDDEN,
+                'Insufficient permission to respond based on Inquiry parameters.'
```

Comment:
`User does not have permission to respond to the inquiry.` is more clear...

Reply:
I agree it's a bit wordy; I was trying to distinguish that this message was not coming from RBAC.

Reply:
Improved in ae4a81ad4